kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -0,0 +1,300 @@
+        : BaseType(rOther)
+        , mpNurbsSurface(rOther.mpNurbsSurface)
+        , mpNurbsCurve(rOther.mpNurbsCurve)
+    {
+    }
+
+    /// Copy constructor from a geometry with different point type.
+    template<class TOtherCurveContainerPointType, class TOtherSurfaceContainerPointType> NurbsCurveOnSurfaceGeometry(
+        NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TCurveContainerPointType, TOtherSurfaceContainerPointType> const& rOther)
+        : BaseType(rOther)
+        , mpNurbsSurface(rOther.mpNurbsSurface)
+        , mpNurbsCurve(rOther.mpNurbsCurve)
+    {
+    }
+
+    /// Destructor
+    ~NurbsCurveOnSurfaceGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    NurbsCurveOnSurfaceGeometry& operator=(const NurbsCurveOnSurfaceGeometry& rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    /**
+     * @brief Assignment operator for geometries with different point type.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    template<class TOtherCurveContainerPointType, class TOtherSurfaceContainerPointType>
+    NurbsCurveOnSurfaceGeometry& operator=(
+        NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TOtherCurveContainerPointType, TOtherSurfaceContainerPointType> const & rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /*typename BaseType::Pointer Create(
+        PointsArrayType const& ThisPoints) const override
+    {
+        return Kratos::make_shared<NurbsCurveOnSurfaceGeometry>(ThisPoints);
+    }*/
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /** 
+    * @brief This method maps from dimension space to working space and computes the
+    *        number of derivatives at the dimension parameter.
+    * From ANurbs library (https://github.com/oberbichler/ANurbs)
+    * @param LocalCoordinates The local coordinates in dimension space
+    * @param Derivative Number of computed derivatives
+    * @return std::vector<array_1d<double, 3>> with the coordinates in working space
+    * @see PointLocalCoordinates
+    */
+    std::vector<CoordinatesArrayType> GlobalDerivatives(
```

Comment:
void GlobalSpaceCoordinates(...) const override

and maybe put it after the GlobalCoordinates to have the same order as in the master. In #5703 are made the same restructuring as we should do here